addons/mail/models/discuss/discuss_channel_rtc_session.py

```diff
@@ -77,6 +88,23 @@ def _gc_inactive_sessions(self):
         self.search(self._inactive_rtc_session_domain()).unlink()
 
     def action_disconnect(self):
+        session_ids_by_channel_by_url = defaultdict(lambda: defaultdict(list))
+        for rtc_session in self:
+            sfu_channel_uuid = rtc_session.channel_id.sfu_channel_uuid
+            url = rtc_session.channel_id.sfu_server_url
+            if sfu_channel_uuid and url:
+                session_ids_by_channel_by_url[url][sfu_channel_uuid].append(rtc_session.id)
+        key = discuss.get_sfu_key(self.env)
+        if key:
+            for url, session_ids_by_channel in session_ids_by_channel_by_url.items():
+                try:
+                    requests.post(
+                        url + '/v1/disconnect',
+                        data=jwt.sign({'sessionIdsByChannel': session_ids_by_channel}, key=key, ttl=20, algorithm=jwt.ALGORITHM.HS256),
+                        timeout=3
+                    )
+                except (requests.exceptions.HTTPError, requests.exceptions.ConnectionError, requests.exceptions.Timeout):
+                    _logger.warning("Could not disconnect sessions at sfu server %s", url)
         self.unlink()
```

Comment:
Isn't it better to have also this code inside `unlink`?
Why do we even have `action_disconnect` method that does just `unlink` in the first place?

Reply:
In most cases, disconnections are caused by the client, so the disonnection from the sfu is either clean (the client leaves intentionally) or a timeout (the client is not responsive). It is only in the case when the disconnect is "forced" by the action that we have to assume that the client may not be cooperative.

There is an `action_disconnect` because in earlier iterations of the code, there was no overload of `unlink` so `action_disconnect` was doing more.